### ISSUE TITLE
Title misplaced when containing a footnote in the margin (HTML format)

### ISSUE THREAD

**Issue by sun123zxy** [quarto-dev/quarto-cli#7158]
### Bug description

In HTML format, when `reference-location` is set to `margin`, a title containing a footnote is placed improperly as the following snapshot shows.

Probably related: #7153 

### Steps to reproduce

`````qmd
---
title: "Test"
reference-location: margin
---

## A Title With a Footnote^[This is a footnote in a title.]

Blah.

## A Title Without a Footnote

Blah.
`````

### Expected behavior

Should have nothing different from a normal title.

### Actual behavior

![image](https://github.com/quarto-dev/quarto-cli/assets/44888975/7e1656fb-1f1e-486f-b695-df52cdec7e2d)

### Your environment

- IDE: VSCode
- OS: Windows 11

### Quarto check output

```bash
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.1: OK
      Dart Sass version 1.55.0: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.3.340
      Path: C:\Program Files\Quarto\bin
      CodePage: 936

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....(None)

      Unable to locate an installed version of Python 3.
      Install Python 3 from https://www.python.org/downloads/

[>] Checking R installation...........(None)

      Unable to locate an installed version of R.
      Install R from https://cloud.r-project.org/
```

**Comment by mcanouil** [comment]
Could you upgrade your Quarto version to at least the latest stable version which is 1.3.450?

Note that on the current development version I cannot reproduce this.
<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/e896f5f3-9a38-4a56-8b0c-b6371a2982a4">

**Comment by sun123zxy** [comment]
Sorry for the late reply. I've upgrade my Quarto version to 1.3.450 and the problem still exists.

Here is the HTML produced:

```html
<section id="a-title-with-a-footnote" class="level2 page-columns page-full">
<h2 class="anchored page-columns page-full" data-anchor-id="a-title-with-a-footnote">A Title With a Footnote<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a><div class="no-row-height column-margin column-container"><li id="fn1"><p><sup>1</sup>&nbsp;This is a footnote in a title.</p></li></div></h2>
<p>Blah.</p>
</section>
<section id="a-title-without-a-footnote" class="level2">
<h2 class="anchored" data-anchor-id="a-title-without-a-footnote">A Title Without a Footnote</h2>
<p>Blah.</p>
</section>
```

> Note that on the current development version I cannot reproduce this.

Since Quarto 1.4 is overhauling the crossref system, maybe it is influenced by some changes? Comparing the HTML files may give some idea about this.

**Comment by mcanouil** [comment]
I don't think that is worth a backport to 1.3 in this context as 1.4 is based on a complete different cross-reference system.

For short, if you can work with 1.4, I suggest you do.
Unless Quarto maintainers think it's worth figuring out the issue in 1.3 and fix this only in 1.3.

**Comment by sun123zxy** [comment]
You are right. Since in 1.4 things are different and this issue can't be reproduced, whether it is fixed in 1.3 is of less importance. I can accept that.

**Comment by mcanouil** [comment]
I am going and closed this as solved.
Feel free to reopen if there is any issue related to this.
